ipn/prefs.go

```diff
@@ -956,8 +967,13 @@ func PrefsFromBytes(b []byte, base *Prefs) error {
 	if len(b) == 0 {
 		return nil
 	}
-
-	return json.Unmarshal(b, base)
+	if err := json.Unmarshal(b, base); err != nil {
+		return err
+	}
+	if base.Sync == opt.ExplicitlyUnset {
```

Comment:
Isn't it too late to check this? I think we've already clobbered the previous value, if there was one.

Reply:
Oh wow, I wonder if that doc comment is even accurate still, or if that promise is even needed.

Let me audit.